### ISSUE TITLE
No CaffeConfig.cmake.  Can't configured on Ubuntu without root access.

### ISSUE THREAD

**Issue by korabelnikov** [HyeonwooNoh/caffe#8]
CMake Warning at CMakeLists.txt:33 (find_package):
   By not providing "FindCaffe.cmake" in CMAKE_MODULE_PATH this project has
   asked CMake to find a package configuration file provided by "Caffe", but
   CMake did not find one.

   Could not find a package configuration file provided by "Caffe" with any of
   the following names:

```
 CaffeConfig.cmake
 caffe-config.cmake
```

   Add the installation prefix of "Caffe" to CMAKE_PREFIX_PATH or set
   "Caffe_DIR" to a directory containing one of the above files.  If "Caffe"
   provides a separate development package or SDK, be sure it has been
   installed.

**Comment by HongjieLiu** [comment]
Hi korabelnikov, I also met this problem. Did you find a way to solve it? Thanks!

**Comment by williford** [comment]
You can install without root access using the official Make configuration:
http://caffe.berkeleyvision.org/installation.html#compilation

**Comment by autocyz** [comment]
Did you make caffe using cmake: http://caffe.berkeleyvision.org/installation.html#compilation
If you make caffe just using make,the caffe project won't  create FindCaffe.cmake.

**Comment by williford** [comment]
I have implemented most of this fork of Caffe to a recent version of Caffe here:
https://github.com/williford/segmentation-caffe

CMake should work, however, it requires a different syntax for the prototxt, so the model definitions would need to be upgraded.

**Comment by navneet1083** [comment]
While compiling some other project, i also get the similar error. Anyone has succeeded ?

Where can i find caffconfig.cmake or do i need to create one ?

CMake Error at CMakeLists.txt:17 (find_package):
  By not providing "FindCaffe.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Caffe", but
  CMake did not find one.

  Could not find a package configuration file provided by "Caffe" with any of
  the following names:

```
CaffeConfig.cmake
caffe-config.cmake
```

  Add the installation prefix of "Caffe" to CMAKE_PREFIX_PATH or set
  "Caffe_DIR" to a directory containing one of the above files.  If "Caffe"
  provides a separate development package or SDK, be sure it has been
  installed.

**Comment by autocyz** [comment]
@navneet1083 
After you clone caffe ,you should make  caffe by cmake:
cd caffe 
`cd caffe`
`mkdir build`
`cd build`
`cmake ..`
`make all`
`make install`
`make runtest`
After do this ,your project will find CaffeConfig.cmake

Hope this will be helpful to you!

**Comment by autocyz** [comment]
> mnae 
sd a d

**Comment by XuXiaoXiao1992** [comment]
@autocyz When I make caffe by cmake, I get the error. Do you have some suggestion? Thank you! 

CMake Error at src/caffe/proto/cmake_install.cmake:40 (FILE):
file INSTALL cannot find
"/home/will/deepLearning/DecoupledNet/caffe-master/build/src/caffe/proto/caffe_pretty_print_pb2.py".
Call Stack (most recent call first):
src/caffe/cmake_install.cmake:41 (INCLUDE)
cmake_install.cmake:42 (INCLUDE)

make: **\* [install] Error 1

**Comment by korabelnikov** [comment]
@williford your fork of caffe named like my main research. What you changed and what kind of segmentation you are perfromming? Thanks

**Comment by korabelnikov** [comment]
@HongjieLiu did you solved you problem? if so, feel free for reopen issue

**Comment by williford** [comment]
@korabelnikov It basically "rebases" this fork of Caffe with the main fork of Caffe. I can release an updated DecoupledNet fork for this new version of Caffe if you are interested, just open an issue on https://github.com/williford/segmentation-caffe. I haven't worked with the DeconvNet model, so I will not be creating a fork of that.

One of the main benefits of using my branch is the ability to use the BatchNorm layer that is included in Caffe. The batch normalization implemented in this branch of Caffe seems to be buggy and unstable.
